redis_array_impl.c

```diff
@@ -466,7 +466,7 @@ ra_find_node(RedisArray *ra, const char *key, int key_len, int *out_pos TSRMLS_D
     if(!out) return NULL;
 
     if(ra->z_dist) {
-        if (!ra_call_distributor(ra, key, key_len, &pos TSRMLS_CC)) {
+        if (!ra_call_distributor(ra, out, out_len, &pos TSRMLS_CC)) {
```

Comment:
But if it's correct behavior to pass original key you should move call `ra_extract_key` to else section and remove `efree` from this one